### PR TITLE
Rely on pulpcore to set up Apache fragments

### PR DESCRIPTION
**Pull request by ekohl**
This uses https://github.com/theforeman/puppet-pulpcore/pull/105 to avoid defining all parts in this module. This means all functionality can be tested standalone in puppet-pulpcore as atomic units.

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl could you rebase this to allow combined testing ?

File: manifests/init.pp [hunk 1]
```diff
@@ -372,32 +373,13 @@
...
+      redirect_legacy => $proxy_pulp_isos_to_pulpcore,
     }
-
-    if $proxy_pulp_yum_to_pulpcore {
-      foreman::config::apache::fragment { 'pulpcore-yum':
-        content     => template('foreman_proxy_content/pulpcore-yum-apache.conf.erb'),
-        ssl_content => template('foreman_proxy_content/pulpcore-yum-apache.conf.erb'),
-      }
+    class { 'pulpcore::plugin::rpm':
+      redirect_legacy => $proxy_pulp_yum_to_pulpcore,
```
Comment (ehelms) [inline manifests/init.pp:380]: These should now be `use_pulp2_content_route`

File: .fixtures.yml [hunk 2]
```diff
@@ -19,7 +19,9 @@ fixtures:
     certs:         "https://github.com/theforeman/puppet-certs.git"
     qpid:          "https://github.com/theforeman/puppet-qpid.git"
     pulp:          "https://github.com/theforeman/puppet-pulp.git"
-    pulpcore:      "https://github.com/theforeman/puppet-pulpcore.git"
+    pulpcore:
+      repo: "https://github.com/ekohl/puppet-pulpcore.git"
```
Comment (ekohl) [inline .fixtures.yml:23]: This obviously needs to be converted in an updated minimum version in metadata.json.

**Comment by ekohl** [comment]
Rebased. First commit is pretty unchanged other solving than the merge conflict and adding fixtures.yml. The second commit is an untested change that I think adds support for running Pulpcore on a content proxy.

**Comment by ekohl** [comment]
The initial support to run without Foreman is here. I suspect that on EL8 (or without Pulp 2 on EL7) it will fail because it doesn't set the certificates yet. Currently running an EL7 pipeline.

File: manifests/init.pp [hunk 3]
```diff
@@ -331,13 +334,33 @@
...
+      $apache_http_vhost = 'foreman'
+      $apache_https_vhost = 'foreman-ssl'
+      Class['foreman::config::apache'] -> Class['pulpcore::apache']
+    } elsif $pulp and $pulp::manage_httpd {
+      $servername = $facts['networking']['fqdn']
+      $priority = '05'
+      $apache_http_vhost = 'pulp-http'
+      $apache_https_vhost = 'pulp-https'
+      Class['pulp::apache'] -> Class['pulpcore::apache']
+    } else {
```
Comment (ehelms) [inline manifests/init.pp:351]: This else is stand-alone Pulpcore? aka a mirror?

Comment (ekohl) [reply manifests/init.pp:351]: Yes, but without a Pulp 2 install. I haven't tested this branch yet, but I think you'll hit this if you spin up an EL8 proxy (if we provide the correct parameters).

**Comment by ehelms** [comment]
Looks like maybe something isn't entirely configured properly on a mirror:

```
    # (in test file fb-proxy.bats, line 26)
    #   `hammer capsule content synchronize --id=$PROXY_ID' failed with status 70
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.0/1, 0%, elapsed: 00:00:00
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.05/1, 5%, 0.0/s, elapsed: 00:00:02, ETA: 00:00:41
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:04, ETA: 00:00:39
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:06, ETA: 00:00:59
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:08, ETA: 00:01:19
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:10, ETA: 00:01:39
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:13, ETA: 00:01:59
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:15, ETA: 00:04:00
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:17
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:19
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:21
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.1/1, 10%, 0.0/s, elapsed: 00:00:23
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:26, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:28, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:30, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:32, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:34, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:37, ETA: 00:03:02
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:39
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:41
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:43
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:46
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:50
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:52
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:54
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 0.16/1, 16%, 0.0/s, elapsed: 00:00:56
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e running: 1.0/1, 100%, 0.0/s, elapsed: 00:00:58
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e warning: 1.0/1, 100%, 0.0/s, elapsed: 00:01:00
    # Task 207f1083-c916-41c5-96e0-518d0d19b29e warning: 1.0/1, 100%, 0.0/s, elapsed: 00:01:00
    # Error: RPM1004: Error retrieving metadata: Error 'Proxy Error' for https://pipeline-katello-server-nightly-centos7.war.example.com/pulp/repos/Test_Organization/Library/Test_CV/custom/Test_Product/Zoo/repodata/76b2c1c5b8ac8a1b9a88c15c2fe82e23cacbfb864ecf55e6b22da879d8cb3ce2-updateinfo.xml.gz.
    not ok 4 content is available from proxy
    # (in test file fb-proxy.bats, line 31)
    #   `URL2="http://${PROXY_HOSTNAME}/pulp/repos/${ORGANIZATION_LABEL}/Library/${CONTENT_VIEW_LABEL}/custom/${PRODUCT_LABEL}/${YUM_REPOSITORY_LABEL}/Packages/w/walrus-0.71-1.noarch.rpm"' failed
    #   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    #                                  Dload  Upload   Total   Spent    Left  Speed
    #   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    # curl: (22) The requested URL returned error: 404 Not Found
    #   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    #                                  Dload  Upload   Total   Spent    Left  Speed
    #   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    # curl: (22) The requested URL returned error: 404 Not Found
```

**Comment by ekohl** [comment]
I noticed the same thing, but couldn't track it down yet.

**Review by ehelms — APPROVED** [review]

File: .fixtures.yml (hunk 2 above)
Comment (ekohl) [reply .fixtures.yml:23]: This needed to be changed before it was merged.

Comment (ekohl) [reply .fixtures.yml:23]: (pushed 45e251bd10715cbc0d8fce4be67b0897e269b503 directly to master)

**Pull request merged**